addons/account/models/account_move_line.py

```diff
@@ -1847,94 +1840,483 @@ def get_accounting_rate(vals):
+                    'reconciled_line_ids': [Command.link(aml.id) for aml in amls],
+                })
+                full_reconcile_fulL_batch_index.append(fulL_batch_index)
+
+        self.env['account.full.reconcile']\
+            .with_context(
+                skip_invoice_sync=True,
+                skip_invoice_line_sync=True,
+                skip_account_move_synchronization=True,
+                check_move_validity=False,
+            )\
+            .create(full_reconcile_values_list)
+
+        # === Cash basis rounding autoreconciliation ===
+        # In case a cash basis rounding difference line got created for the transition account, we reconcile it with the corresponding lines
+        # on the cash basis moves (so that it reaches full reconciliation and creates an exchange difference entry for this account as well)
+        for fulL_batch_index, full_batch in enumerate(full_batches):
+            if not full_batch.get('caba_lines_to_reconcile'):
+                continue
+
+            caba_lines_to_reconcile = full_batch['caba_lines_to_reconcile']
+            exchange_move = full_batch['exchange_move']
+            for (dummy, account, repartition_line), amls_to_reconcile in caba_lines_to_reconcile.items():
+                if not account.reconcile:
+                    continue
+
+                exchange_line = exchange_move.line_ids.filtered(
+                    lambda l: l.account_id == account and l.tax_repartition_line_id == repartition_line
+                )
+
+                (exchange_line + amls_to_reconcile)\
+                    .filtered(lambda l: not l.reconciled)\
+                    .reconcile()
+
+        all_amls._reconcile_post_hook(pre_hook_data)
 
     def _create_reconciliation_partials(self):
         '''create the partial reconciliation between all the records in self
          :return: A recordset of account.partial.reconcile.
         '''
         partials_vals_list, exchange_data = self._prepare_reconciliation_partials([
             {
-                'record': line,
-                'balance': line.balance,
-                'amount_currency': line.amount_currency,
+                'aml': line,
                 'amount_residual': line.amount_residual,
                 'amount_residual_currency': line.amount_residual_currency,
-                'company': line.company_id,
-                'currency': line.currency_id,
-                'date': line.date,
             }
             for line in self
         ])
         partials = self.env['account.partial.reconcile'].create(partials_vals_list)
 
         # ==== Create exchange difference moves ====
-        for index, exchange_vals in exchange_data.items():
-            partials[index].exchange_move_id = self._create_exchange_difference_move(exchange_vals)
+        for index, exchange_values in exchange_data.items():
+            partials[index].exchange_move_id = self._create_exchange_difference_move(exchange_values)
```

Comment:
Couldn't this creation part be batched too and assign it right after?

Reply:
@Levizar Probably. This will complexify the code a lot and I would already merge this part first (x11) before challenging more complicated reconciliations